### ISSUE TITLE
Multiselect attributes gathered from child products might be converted to objects instead of arrays during query

### ISSUE THREAD

**Issue by sambolek** [Smile-SA/elasticsuite#533]
### Preconditions

Magento Version : Magento CE 2.1.6

ElasticSuite Version : source - c273b232a8fb3e33344b9501e44f194370262d6f

Environment : Developer

Third party modules : not relevant

### Steps to reproduce

1. Have a multiselect attribute
2. Assign same multiselect attribute, but different values to both bundle and it's child
3. Save product

### Expected result

1. Product saves without elasticsearch errors. 
2. Product displays in drag&drop admin section and frontend

### Actual result

1. Product isn't displayed in drag&drop admin section
2. Product is not visible in categories on frontend
3. An error in elasticsearch is thrown
`MapperParsingException[failed to parse [tech]]; nested: IllegalArgumentException[unknown property [0]];`
`"some_multiselect_attribute":{"0":"479","1":"481","2":"487","3":"489","4":"490","7":"483","9":"1620","10":"488"}`

When adding child data inside:
```
src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/AttributeData.php:63
```

array keys for attributes can be in non-sequential form, because they're filtered through array_unique on `src/module-elasticsuite-catalog/Model/Product/Indexer/Fulltext/Datasource/AttributeData.php:124`
E.g.
`['some_multiselect_attribute'] = [0 => '123', 2 => '456']`

this presents a problem for SmartSerializer, as json_encode is interpreting that array as an object instead of array https://stackoverflow.com/questions/18116987/php-json-encode-strange-behaviour

The solution is, like on other places where we're collecting attribute data, to pass that array to "array_values" function that resets array keys and allows json_encode to correctly interpret attribute data as array.

I'll submit a pull request that resolves this.

**Comment by afoucret** [comment]
PR have been merged into branches master, 2.3.x and 2.4.x

**Comment by sambolek** [comment]
Thanks!
